mpqp/execution/simulated_devices.py

```diff
@@ -0,0 +1,75 @@
+"""To execute a circuit on a noisy simulator that reproduces the noise model of a machine,
+one can use a :class:`SimulatedDevice`. Inheriting from :class:`~mpqp.execution.devices.AvailableDevice`
+"""
+
+# TODO: finish doc
+
+
+from typing import TYPE_CHECKING
+
+from typeguard import typechecked
+
+from mpqp.execution import AvailableDevice
+
+if TYPE_CHECKING:
+    from qiskit_aer.backends.aer_simulator import AerSimulator
+    from qiskit_aer.noise import NoiseModel as Qiskit_NoiseModel
+    from qiskit_ibm_runtime.fake_provider.fake_backend import FakeBackendV2
+
+
+class SimulatedDevice(AvailableDevice):
+    """Abstract class used to define simulators reproducing the noise of a real device."""
+
+    # TODO : comment
+    def is_gate_based(self) -> bool:
+        return True
+
+    def is_simulator(self) -> bool:
+        return True
+
+    def is_noisy_simulator(self) -> bool:
+        return True
+
+    def is_remote(self) -> bool:
+        return False
+
+
+@typechecked
+class AbstractIBMSimulatedDevice(SimulatedDevice):
```

Comment:
add `ABC` as a base

Reply:
do we really need this intermediate level though ?

Reply:
I don't think so, see link I added. If need be I can do it myself

Reply:
I saw the link you added, but I cannot identify where they define any additional method for those enums.
If you can try to make it a single class it would be cool.

Reply:
Trello task created. We will improve that later